File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/api/dto/SessaoResponse.java
package br.ufpr.sept.so2.modules.iam.api.dto;

import br.ufpr.sept.so2.modules.iam.application.ConsultarSessaoUseCase;

import java.util.List;
import java.util.UUID;

public record SessaoResponse(UUID id, boolean mustChangePassword, List<String> authorities) {

    public static SessaoResponse from(ConsultarSessaoUseCase.SessaoAtual sessao) {
        return new SessaoResponse(sessao.id(), sessao.mustChangePassword(), sessao.authorities());
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/api/dto/SessaoResponse.kt
package br.ufpr.sept.so2.modules.iam.api.dto

import br.ufpr.sept.so2.modules.iam.application.ConsultarSessaoUseCase
import br.ufpr.sept.so2.modules.iam.application.MenuLinks
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class SessaoResponse(
    val id: UUID,
    val mustChangePassword: Boolean,
    val authorities: List<String>,
    @get:JsonProperty("_links")
    val links: Map<String, String>,
) {
    companion object {
        fun from(sessao: ConsultarSessaoUseCase.SessaoAtual): SessaoResponse =
            SessaoResponse(
                sessao.id,
                sessao.mustChangePassword,
                sessao.authorities,
                MenuLinks.from(sessao.authorities, sessao.cursoConfigurarId),
            )
    }
}
